android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt

```diff
@@ -72,7 +70,7 @@ class GestureHandlerOrchestrator(
 
   private fun cleanupFinishedHandlers() {
     var shouldCleanEmptyCells = false
-    for (i in gestureHandlersCount - 1 downTo 0) {
+    for (i in gestureHandlers.size - 1 downTo 0) {
```

Comment:
```suggestion
    for (handler in gestureHandlers.reversed()) {
```

Same in other places when iterating in reverse

Reply:
Done in [6957bb1](https://github.com/software-mansion/react-native-gesture-handler/pull/2672/commits/6957bb1387ac38cfad1c24e4c5b5e63333f7c28f)